serving/samples/jib-app-java/README.md

```diff
@@ -0,0 +1,99 @@
+# Jib Sample App
+
+A sample app that demonstrates using [Jib](https://github.com/GoogleContainerTools/jib) to build Java applications
+and running with Knative Serving.
+
+Jib can package any Java application without using the Dockerfile.
+
+This deploys the [Helloworld Java](https://github.com/knative/docs/tree/master/serving/samples/helloworld-java)
+sample app for Knative, and builds without using its Dockerfile.
+
+
+## Prerequisites
+
+* [Install Knative Serving](../../../install/README.md)
+* [Configure Image Push Credentials](../../../build/auth.md#basic-authentication-docker)
+
+## Running
+
+This sample uses the [Jib build
+template](https://github.com/knative/build-templates/tree/master/jib)
+in the [build-templates](https://github.com/knative/build-templates/) repo.
+
+Install the Jib Maven build template:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/knative/build-templates/master/jib/jib-maven.yaml
+```
+
+Then you can deploy this to Knative Serving from the root directory
+by entering the following commands:
+
+```shell
+export REPO="your/registry"
+
+# For example, with Google Container Registry:
+# export REPO="gcr.io/YOUR_PROJECT_ID"
+
+perl -pi -e "s@DOCKER_REPO_OVERRIDE@$REPO@g" sample.yaml
+
+# Create the Kubernetes resources
```

Comment:
I think (check our other examples) that you might want to spell out in a little more detail what's going on here. In particular, I think there is a single Service which specifies a Build using the Jib template. The user doesn't need to have docker or other tools installed, and they end up with:

a. a container they can run anywhere (including on Knative)
b. a running service using the supplied code, with the arguments specified by the Service.

This is pretty cool, so it's worth spelling out a bit more about what's going on here that the user _doesn't_ need to do.

Reply:
i'll call out a bit more about jib

Reply:
I think you should still play this up a bit more -- maybe something like "this build runs on the cluster with a consistent version of Java, regardless of what versions of Java you may have installed locally".

Reply:
Something just before or just after the build would be a good place to point out that this works even without Maven installed, as well.